### PR TITLE
1314058 import macros

### PR DESCRIPTION
**Pull request by escattone**

### CODE REVIEW

**Comment by escattone** [comment]
One thing I don't like yet is what happens when there is a "duplicate macros" error within the FileLoader. Currently, when the pool of Hirelings is started, the pool will infinitely loop on restarting failed hireling processes. I think that error should be caught before the pool is created so that the server fails to start and logs the error.

**Comment by escattone** [comment]
The latest commit ensures that if the FileLoader would throw a "duplicate macros" error, it's detected at startup time, causing the KumaScript service to fail to start.

**Comment by escattone** [comment]
I recommitted the macros using the latest versions as of 2 hours ago or so. I'm giving it my approval for merge and awaiting further review.

File: lib/kumascript/conf.js [hunk 1]
```diff
@@ -33,10 +33,9 @@ var DEFAULT_CONFIG = {
         workerTimeout: 1000 * 60,
         workerMaxJobs: 8,
         workerRetries: 3,
+        template_root_dir: "macros",
         document_url_template:
             "https://developer.mozilla.org/en-US/docs/{path}?raw=1",
-        template_url_template:
-            "https://developer.mozilla.org/en-US/docs/en-US/Template:{name}?raw=1",
```
Comment (jwhitlock) [inline lib/kumascript/conf.js]: Should we remove this now, or wait until we remove the ``HTTPLoader``?

File: macros/FirefoxOSDefaultAPISidebar.ejs [hunk 2]
```diff
@@ -0,0 +1 @@
+REDIRECT <a class="redirect" href="/en-US/docs/Template:FirefoxOSAPISidebar">Template:FirefoxOSDefaultAPISidebar</a>
```
Comment (jwhitlock) [inline macros/FirefoxOSDefaultAPISidebar.ejs]: This is the source for an MDN page that is a redirect to another file. If possible, we should not import these redirects.  One way to detect them is to look for content that starts with ``REDIRECT <a ``

I'm not sure how the live site currently handles these redirects. In other places, the pattern was not a redirect, but instead something like ``ifandmethod.ejs``:

```
<% /* Please use the up to date ifmethod template, this template currently redirects there. */ %>
<%- template('ifmethod', [$0, $1]) %>
```

or the more generic ``HTML.ejs``:

```
<%- template('Template:HTML:Element_Navigation', arguments) %>
```

Comment (Elchi3) [inline macros/FirefoxOSDefaultAPISidebar.ejs]: [0 pages are calling this macro](https://developer.mozilla.org/en-US/search?locale=*&kumascript_macros=FirefoxOSDefaultAPISidebar&topic=none), so it is fine for this macro to not be imported. For other "redirect macros" that are called by pages, we probably need to create a macro that calls through like `<%- template('Template:HTML:Element_Navigation', arguments) %>`. 
(These are good first candidates for writers/content contributors to clean up once we are done importing. Sigh.)

File: lib/kumascript/conf.js (hunk 1 above)
Comment (escattone) [inline lib/kumascript/conf.js]: I'll keep it for now, but fix the URL (remove the second "en-US").

**Pull request merged**